ios/RNCameraRoll.m

```diff
@@ -11,14 +11,36 @@
 #import “React/RCTBridge.h” // Required when used as a Pod in a Swift project
 #endif
 
+@interface AssetData : NSObject
+
+@property PHAsset *asset;
+@property NSString *path;
+@property UIImageOrientation orientation;
+
+- (NSComparisonResult)compare:(AssetData *)otherObject;
+
+@end
+
+@implementation AssetData
+
+- (NSComparisonResult)compare:(AssetData *)otherObject {
+  NSDate *selfDate = [self.asset valueForKey:@"modificationDate"];
+  NSDate *otherDate = [otherObject.asset valueForKey:@"modificationDate"];
+  return [selfDate compare:otherDate];
+}
+@end
+
 @implementation RNCameraRoll
 
 RCT_EXPORT_MODULE();
 
+- (dispatch_queue_t)methodQueue {
+  return dispatch_queue_create("io.textile.CameraRollQueue", DISPATCH_QUEUE_SERIAL);
```

Comment:
Make sure this native module runs on a background thread because `requestLocalPhotos` now blocks while it asynchronously collects results.